pkg/storage/fs/owncloud/owncloud.go

```diff
@@ -263,7 +263,9 @@ func (fs *ocfs) toInternalPath(ctx context.Context, sp string) (ip string) {
 	if fs.c.EnableHome {
 		u := user.ContextMustGetUser(ctx)
 		layout := templates.WithUser(u, fs.c.UserLayout)
-		ip = filepath.Join(fs.c.DataDirectory, layout, "files", sp)
+		// The inner filepath.Join prevents the path from breaking out of
+		// <fs.c.DataDirectory>/<layout>/files/
+		ip = filepath.Join(fs.c.DataDirectory, layout, "files", filepath.Join("/", sp))
```

Comment:
Can you give an example where it would break without the inner join?

Reply:
If `sp` starts with `..` then it would 'break out' of our files directory.
See for example: https://play.golang.org/p/JVfjiNAYCR0

Reply:
Ah cool